docs/resources/server/loottables/index.md

```diff
@@ -0,0 +1,322 @@
+- A **loot pool** is, in essence, a list of loot entries. A loot table can contain multiple loot pools, and each loot pool is rolled independently of the others.
+  - NeoForge allows loot pools to specify a `name` field. This can be used by [GLMs][glm]. If unspecified, this is the hash code of the loot pool, prefixed by `custom#`.
+- A **roll** is a "picking" from a loot table or loot pool. A loot table is always rolled once per in-code call, but loot pools can specify this value to be greater than 1.
+  - A **bonus roll** is an extra roll, calculated using the [**luck value**][luck]. It is specified and calculated independently of the regular rolls amount, and added on top of it after both calculations have completed.
+- A **loot condition** is a condition on a loot entry that must be met for the entry to be further considered, otherwise the entry is treated as non-existent.
+  - Multiple loot conditions can be applied to the same loot entry.
+  - Loot conditions (or lists thereof) can also be applied to a loot pool. If the condition fails, the entire pool is discarded.
+  - This is not to be confused with NeoForge's [conditional data loading][conditions]. Data load conditions may be used in a loot table as well, but only in the root.
+  - For a full list of vanilla loot conditions, see [Item Predicates][itempredicates]. [NeoForge adds two loot conditions as well][neoconditions]. Modders may also add their own loot conditions.
+- A **loot function** is a function that can be applied to the result of a loot entry.
+  - Multiple loot functions can be applied to the same loot entry.
+  - Loot functions (or lists thereof) can also be applied to a loot pool or loot table, affecting all items in that loot pool/loot table.
+  - For a full list of vanilla loot functions, see [Item Modifiers][itemmodifiers]. Modders may also add their own loot functions.
+- The **loot context** is an object passed to the loot table evaluator that may contain additional information for rolling the table, for example an associated block state, block entity, entity, etc.; these are called **loot context parameters**.
+  - A full list of loot context parameters can be found in the `LootContextParams` class.
+- The **loot table type** is a validator for the loot context. For example, the loot table type `barter` is guaranteed to have an entity context for the piglin being bartered with.
+  - The context is validated at data pack load time, printing a log warning if context is used that is not defined by the loot table type.
+  - The loot table type is always optional. No validation will take place if it is not specified.
+  - This is internally called a **loot context parameter set**. The default parameter sets can be found in the `LootContextParamSets` class.
+  - For a full list of loot table types and what context they provide, see [Loot Contexts][lootcontext].
+- A **loot table evaluator** is the piece of Java code rolling the loot table.
```

Comment:
This just looks like a wall of text to me and doesn't help me understand how this links to the codebase. This should be simplified to some degree or have the code equivalents front and center.

Reply:
Yes, this section is due for another rework.